sapi/fpm/fpm/fpm_main.c

```diff
@@ -1996,7 +1996,8 @@ consult the installation file that came with this distribution, or visit \n\
 
 			requests++;
 			if (UNEXPECTED(max_requests && (requests == max_requests))) {
-				fcgi_finish_request(request, 1);
+				fcgi_request_set_keep(request, 0);
+				fcgi_finish_request(request, 0);
```

Comment:
This looks reasonable. Should the same change also be done [here](https://github.com/php/php-src/blob/bdfe07c2e7e71ffd1d6f0cf956612189564f4228/sapi/cgi/cgi_main.c#L2688)?